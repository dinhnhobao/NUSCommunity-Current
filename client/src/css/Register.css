.register {
  background: #35d0ba;
  width: 1140px;
  margin: 0 auto;
  padding: 36px;
}
.register-left {
  margin-top: 48px;
}
.register-left-img {
  margin: 0 auto;
  margin-top: 70px;
  margin-bottom: 10px;
  width: 200px;
  display: block;
}
.register-left-btn {
  border-radius: 24px;
  padding: 5px;
  width: 140px;
  background: #f8f9fa;
  font-weight: bold;
  color: #383d41;
  margin: 0 auto;
  margin-top: 70px;
  margin-bottom: 8px;
  cursor: pointer;
  text-align: center;
  display: block;
  font-weight: bolder;
}
.register-right {
  padding-top: 80px;
  width: 100%;
  background: #f8f9fa;
  border-top-left-radius: 10% 50%;
  border-bottom-left-radius: 10% 50%;
}
.register-right-title {
  text-align: center;
  font-weight: bolder;
  margin-bottom: 46px;
  line-height: 34px;
  font-size: 28px;
  color: #495057;
}
.register-right-form {
  margin: 0 100px;
}
.register-right-form-button {
  display: block;
  width: 400px;
  margin: 0 auto;
  margin-top: 20px;
  margin-bottom: 80px;
}
.alert-style {
  text-align: center;
  color: red;
  font-weight: bold;
}
