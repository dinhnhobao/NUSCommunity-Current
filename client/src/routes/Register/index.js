import React from "react";
import { Link } from "react-router-dom"
import Router from 'next/router'
//import Link from 'next/link'
import { Form, Input, Button, Row, Col } from "antd";
import icon from "../../assets/favicon.ico";
import "./index.css"
import Session from '../../utils/session'

/*
const onFinish = (values) => {
    console.log(values);
    alert("A verification link has been sent to your email");
    window.location.href = "/login";
  };
*/

class Register extends React.Component {
    static async getInitialProps({ req, res }) {

        let props = {
            session: ''
        }

        if (req && req.session) {
            props.session = req.session
        } else {
            props.session = await Session.getSession()
        }

        if (props.session && props.session.loggedin) {
            if (req) {
                res.redirect('/')
            } else {
                Router.push('/')
            }
        }

        return props
    }

    constructor(props) {
        super(props)
        this.state = {
            email: '',
            password: '',
            confirmPassword: '',
            username: '',
            message: null
        }
        this.handleEmailChange = this.handleEmailChange.bind(this)
        this.handlePasswordChange = this.handlePasswordChange.bind(this)
        this.handleConfirmPasswordChange = this.handleConfirmPasswordChange.bind(this)
        this.handleUsernameChange = this.handleUsernameChange.bind(this)
        this.handleSignUp = this.handleSignUp.bind(this)
    }

    handleEmailChange(event) {
        this.setState({
            email: event.target.value.trim()
        })
    }

    handlePasswordChange(event) {
        this.setState({
            password: event.target.value.trim()
        })
    }

    handleConfirmPasswordChange(event) {
        this.setState({
            confirmPassword: event.target.value.trim()
        })
    }

    handleUsernameChange(event) {
        this.setState({
            username: event.target.value.trim()
        })
    }

    handleSignUp(event) {
        event.preventDefault()

        this.setState({
            message: null
        })

        if (!this.state.email || !this.state.password || !this.state.confirmPassword || !this.state.username) {
            this.setState({
                message: 'All fields are required!'
            })

            return
        }

        if (this.state.password !== this.state.confirmPassword) {
            this.setState({
                message: 'Password did not match!'
            })

            return
        }

        let data = {
            email: this.state.email,
            password: this.state.password,
            username: this.state.username
        }

        fetch('auth/signup', {
            method: 'POST',
            body: JSON.stringify(data),
            headers: {
                'Content-Type': 'application/json'
            }
        })
            .then(res => res.json())
            .then(response => {
                if (response.message) {
                    this.setState({
                        message: response.message
                    })
                } else if (response.email) {
                    this.setState({
                        message: "A verification link has been sent to " + response.email
                    })
                    //alert("A verification link has been sent to " + response.email);
                    //Router.push('/check-email?email=' + response.email)
                } else {
                    this.setState({
                        message: 'Unknown Error!'
                    })
                }
            })
            .catch(error => {
                console.error('Error:', error)
                this.setState({
                    message: 'Request Failed!'
                })
            })
    }

    render() {
        const alert = (this.state.message === null) ? <div /> : <div className="alert-style" role="alert">{this.state.message}</div>
        return (
            <div style={{ paddingTop: 36 }}>
                <div className="register">
                    <Row gutter={16}>
                        <Col span={6}>
                            <div className="register-left">
                                <img className="register-left-img" src={icon} alt="" />
                                <Link className="register-left-btn" to="/login">
                                    Login
                                </Link>
                            </div>
                        </Col>
                        <Col span={18}>
                            <div className="register-right">
                                <div className="register-right-title">Sign up form</div>
                                <Form
                                    name="normal_register"
                                    className="register-right-form"
                                >
                                    <Form.Item
                                        name="email"
                                        rules={[
                                            {
                                                required: true,
                                                message: "Please enter a valid NUS email!",
                                            },
                                        ]}
                                    >
                                        <Input placeholder="NUS email" name="email" addonAfter="@u.nus.edu" value={this.state.email} onChange={this.handleEmailChange} />
                                    </Form.Item>
                                    <Form.Item
                                        name="password"
                                        rules={[
                                            {
                                                required: true,
                                                message: "Please input your password!",
                                            },
                                        ]}
                                    >
                                        <Input type="password" name="password" placeholder="Password" value={this.state.password} onChange={this.handlePasswordChange} />
                                    </Form.Item>
                                    <Form.Item
                                        name="confirm"
                                        dependencies={["password"]}
                                        rules={[
                                            {
                                                required: true,
                                                message: "Please input your password again！",
                                            },
                                            ({ getFieldValue }) => ({
                                                validator(rule, value) {
                                                    if (!value || getFieldValue("password") === value) {
                                                        return Promise.resolve();
                                                    }
                                                    // eslint-disable-next-line prefer-promise-reject-errors
                                                    return Promise.reject(
                                                        "The confirm password does not match with the password!"
                                                    );
                                                },
                                            }),
                                        ]}
                                    >
                                        <Input type="password" name="confirmPassword" placeholder="Confirm Password" value={this.state.confirmPassword} onChange={this.handleConfirmPasswordChange} />
                                    </Form.Item>
                                    <Form.Item
                                        name="username"
                                        rules={[
                                            {
                                                required: true,
                                                message: "Do not leave username blank!",
                                            },
                                        ]}
                                    >
                                        <Input placeholder="Username" name="username" value={this.state.username} onChange={this.handleUsernameChange} />
                                    </Form.Item>
                                    {alert}
                                    <Form.Item>
                                        <Button
                                            type="submit"
                                            onClick={this.handleSignUp}
                                            className="register-right-form-button"
                                            size="large"
                                        >
                                            Register
                                        </Button>
                                    </Form.Item>
                                </Form>
                            </div>
                        </Col>
                    </Row>
                </div>
            </div>
        );
    };
}

export default Register;
