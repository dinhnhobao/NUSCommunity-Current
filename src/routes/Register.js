import React from "react";
import { Link } from "react-router-dom";
import { Form, Input, Button, message, Row, Col } from "antd";
import icon from "../assets/favicon.ico";
import "./Register.css";


const Register = () => {
  // 注册账号数据处理
  const onFinish = (values) => {
    console.log(values);
    message.success("register success~");
    window.location.href = "/login";
  };


  return (
    <div style={{ paddingTop: 36 }}>
      <div className="register">
        <Row gutter={16}>
          <Col span={6}>
            <div className="register-left">
              <img className="register-left-img" src={icon} alt="" />
              <Link className="register-left-btn" to="/login">
                Login
              </Link>
            </div>
          </Col>
          <Col span={18}>
            <div className="register-right">
              <div className="register-right-title">Sign up form</div>
              <Form
                name="normal_register"
                className="register-right-form"
                onFinish={onFinish}
                action="auth/sign-up"
                method="POST"
              >
                <Form.Item
                  name="email"
                  rules={[
                    {
                      required: true,
                      message: "Please enter a valid NUS email!",
                    },
                  ]}
                >
                  <Input placeholder="NUS email" name="email" addonAfter="@u.nus.edu" />
                </Form.Item>
                <Form.Item
                  name="password"
                  rules={[
                    {
                      required: true,
                      message: "Please input your password!",
                    },
                  ]}
                >
                  <Input type="password" name="password" placeholder="Password" />
                </Form.Item>
                <Form.Item
                  name="confirm"
                  dependencies={["password"]}
                  rules={[
                    {
                      required: true,
                      message: "Please input your password again！",
                    },
                    ({ getFieldValue }) => ({
                      validator(rule, value) {
                        if (!value || getFieldValue("password") === value) {
                          return Promise.resolve();
                        }
                        // eslint-disable-next-line prefer-promise-reject-errors
                        return Promise.reject(
                          "The confirm password does not match with the password!"
                        );
                      },
                    }),
                  ]}
                >
                  <Input type="password" placeholder="Confirm Password" />
                </Form.Item>
                <Form.Item
                  name="username"
                  rules={[
                    {
                      required: true,
                      message: "Do not leave username blank!",
                    },
                  ]}
                >
                  <Input placeholder="Username" type="text" name="username" />
                </Form.Item>
                <Form.Item>
                  <Button
                    htmlType="submit"
                    type="primary"
                    className="register-right-form-button"
                    ghost
                    size="large"
                  >
                    Register
                  </Button>
                </Form.Item>
              </Form>
            </div>
          </Col>
        </Row>
      </div>
    </div>
  );
};

export default Register;
