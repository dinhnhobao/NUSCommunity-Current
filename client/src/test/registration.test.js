import React from 'react';
import {shallow} from 'enzyme';
import Register from '../routes/Register/index';

const setUp = (props={}) => {
    const component = shallow(<Register {...props} />);
    return component;
}

const findByTestAttr = (component, attr) => {
    const wrapper = component.find(`[data-test='${attr}']`);
    return wrapper;
}

describe('Register Component', () => {
    let component;
    beforeEach(() => {
        component = setUp();
    })

    it('Should render correctly', () => {
        const wrapper =  findByTestAttr(component, 'registerComponent');
        expect(wrapper).toMatchSnapshot();
    });

    it('Should render without errors', () => {
        const wrapper =  findByTestAttr(component, 'registerComponent');
        expect(wrapper.length).toEqual(1);       
    });

    describe('Registration Form', () => {
        it('Fill out email in the form', () => {
            const wrapper = shallow(<Register/>);
            let emailInput = wrapper.find('Input').first();
            emailInput.simulate('change', { 
                target: {value: 'e0407663'},
            })
            emailInput = wrapper.find('Input').first();
            expect(emailInput.props().value).toEqual('e0407663')
        })

        it('Fill out username in the form', () => {
            const wrapper = shallow(<Register/>);
            let usernameInput = wrapper.find('Input').last();
            usernameInput.simulate('change', { 
                target: {value: 'Anonymous'},
            })
            usernameInput = wrapper.find('Input').last();
            expect(usernameInput.props().value).toEqual('Anonymous')
        })
    })

    describe('Button', () => {
        it('Should render Button with click handle', () => {
            const wrapper =  findByTestAttr(component, 'buttonComponent');
            expect(wrapper.length).toEqual(1);
        })
    })
})