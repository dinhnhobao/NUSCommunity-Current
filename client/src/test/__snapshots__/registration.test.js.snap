// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Component Should render correctly 1`] = `ShallowWrapper {}`;
